src/bpftrace.cpp

```diff
@@ -87,7 +88,8 @@ int BPFtrace::add_probe(ast::Probe &p)
     }
 
     std::vector<std::string> attach_funcs;
-    if (attach_point->need_expansion && has_wildcard(attach_point->func))
+    bool underspecified_usdt_probe = (probetype(attach_point->provider) == ProbeType::usdt  && attach_point->ns.empty());
```

Comment:
If it's a USDT probe and the namespace wasn't given, it's an underspecified probe and we need to do expansion / treat it like a wildcard.